src/simulation/LoadGenerator.cpp

```diff
@@ -221,591 +143,459 @@ maybeAdjustRate(double target, double actual, uint32_t& rate, bool increaseOk)
-        if (acceptablePathExtension(from, ledgerNum, path, mm))
-        {
-            assert(mm->mBuyCredit == issuer);
-            return mm;
-        }
-    }
-    return nullptr;
+    TestAccountPtr to, from;
+    uint64_t amount = 1;
+    std::tie(from, to) = pickAccountPair(numAccounts, ledgerNum, sourceAccount);
+    vector<Operation> paymentOps = {
+        txtest::payment(to->getPublicKey(), amount)};
+    TxInfo tx = TxInfo{from, paymentOps};
+
+    return tx;
 }
 
 void
-randomPathWalk(LoadGenerator::AccountInfoPtr from, uint32_t ledgerNum,
-               std::vector<LoadGenerator::AccountInfoPtr>& path,
-               LoadGenerator::AccountInfoPtr& to)
+LoadGenerator::handleFailedSubmission(TestAccountPtr sourceAccount,
+                                      Herder::TransactionSubmitStatus status,
+                                      TransactionResultCode code)
 {
-    auto issuer =
-        (path.empty() ? rand_element(from->mTrustLines).mIssuer : path.back());
-
-    auto mm = pickMarketMakerForIssuer(from, ledgerNum, path, issuer);
-    if (mm && rand_flip() && path.size() < 5)
-    {
-        // We have a market maker -- mm is buying 'issuer' credits -- and we
-        // want to let mm buy it and sell credit that someone else trusts;
-        // we then see about extending the walk from that someone.
-        assert(mm->mSellCredit);
-        size_t i = mm->mSellCredit->mTrustingAccounts.size();
-        while (i-- != 0)
-        {
-            auto maybeTo = rand_element(mm->mSellCredit->mTrustingAccounts);
-            if (maybeTo != mm &&
-                acceptablePathExtension(from, ledgerNum, path, maybeTo))
-            {
-                path.push_back(issuer);
-                to = maybeTo;
-                randomPathWalk(from, ledgerNum, path, to);
-                return;
-            }
-        }
-    }
-    // No market-maker, just find a destination that can accept credits from the
-    // issuer we've picked.
-    size_t i = issuer->mTrustingAccounts.size();
-    while (i-- != 0)
+    // Note that if transaction is a DUPLICATE, its sequence number is
+    // incremented on the next call to execute.
+    if (status == Herder::TX_STATUS_ERROR && code == txBAD_SEQ)
     {
-        auto maybeTo = rand_element(issuer->mTrustingAccounts);
-        if (maybeTo != from)
+        if (!loadAccount(sourceAccount, mApp.getDatabase()))
         {
-            to = maybeTo;
-            path.push_back(issuer);
-            break;
+            CLOG(ERROR, "LoadGen")
+                << "Unable to reload account " << sourceAccount->getAccountId();
         }
     }
 }
 
-LoadGenerator::AccountInfoPtr
-LoadGenerator::pickRandomPath(LoadGenerator::AccountInfoPtr from,
-                              uint32_t ledgerNum,
-                              std::vector<LoadGenerator::AccountInfoPtr>& path)
+bool
+LoadGenerator::checkAccountSynced(std::vector<TestAccountPtr>& result,
```

Comment:
I think we could just make std::vector<TestAccountPtr> return type and result result.
RVO will take care of copying and having that gives us enough information to get the `synced` value as well.